addons/bus/models/bus_presence.py

```diff
@@ -31,6 +33,21 @@ class BusPresence(models.Model):
 
     @api.model
     def update(self, inactivity_period):
+        try:
+            self.with_context(MAX_TRIES_ON_CONCURRENCY_FAILURE=0)._update(inactivity_period)
+            # Commit on success.
+            # This method is called in _poll method and cursor is closed right after.
+            # See bus/controllers/main.py
+            self.env.cr.commit()
+            return True
+        except OperationalError as e:
+            if e.pgcode in PG_CONCURRENCY_ERRORS_TO_RETRY:
+                return False
+            else:
+                raise
```

Comment:
why not like in the `check` function below?

```suggestion
            if e.pgcode not in PG_CONCURRENCY_ERRORS_TO_RETRY:
                raise
```

why should `update` return a boolean value instead of `None` like before?

Reply:
Returning boolean is not really needed. Like returning `True` in `write` method :wink: